assets/react-hooks/index.js

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useLayoutEffect, useCallback } from '@wordpress/element';
+
+/**
+ * Hook for event listeners.
+ *
+ * @param {string}   eventName Event name to be attached.
+ * @param {Function} handler   Handler function to the event.
+ * @param {Array}    deps      Dependencies of the handler function.
+ * @param {Object}   element   Element to attach the event. Default is `window`.
+ */
+export const useEventListener = (
+	eventName,
+	handler,
+	deps,
+	element = window
+) => {
+	const handlerCallback = useCallback( handler, deps );
+
+	useEffect( () => {
+		const evtArgs = [ eventName, handlerCallback, false ];
+		element.addEventListener( ...evtArgs );
+
+		return () => {
+			element.removeEventListener( ...evtArgs );
+		};
+	}, [ eventName, handlerCallback, element ] );
+};
+
+/**
+ * Apply fullscreen view by hiding wp-admin elements via CSS.
+ *
+ * Allows setting additional classes on the body element.
+ * Fullscreen and classes are added when the component is mounted, and removed when unmounted.
+ *
+ * @param {string[]} bodyClasses Additional classes to be set.
+ */
+export const useFullScreen = ( bodyClasses = [] ) => {
+	const classes = [ ...bodyClasses, 'sensei-wp-admin-fullscreen' ];
```

Comment:
@yscik I moved the `useFullScreen` to here. So we can create a file with all the useful hooks that we can use in other parts of the project. Does it make sense to you?

Reply:
We should probably rename it to something like `useWpAdminFullscreen` if its outside the onboarding folder, as it is somewhat specific to `/wp-admin` right now. 
Also, I think it would be cleaner for each hook to be in its dedicated file.

Reply:
Hum! Makes totally sense! Name updated and changed the structure to be one hook per file and I create an index, so we can import easily when we'll use more than one hook per file.

Does it sound good to you? https://github.com/Automattic/sensei/pull/3085/commits/2a154a98ebc9fa18931704aa723490843a75da85